src/wm_info_provider/hyprland.rs

```diff
@@ -107,8 +106,9 @@ struct Ipc {
 
 impl Ipc {
     fn new(his: &str) -> io::Result<Self> {
-        let sock1_path = format!("/tmp/hypr/{his}/.socket.sock");
-        let sock2_path = format!("/tmp/hypr/{his}/.socket2.sock");
+        let xdgrd = std::env::var("XDG_RUNTIME_DIR").expect("$XDG_RUNTIME_DIR is not set");
```

Comment:
I would prefer not to use `.expect()`. Instead `.map_err()`, create an `io::Error` and propagate it. Or, since the error from `Ipc::new()` is ignored anyway, you may change the return type of this function to `Opton<Self>`, in which case you will be able to write `.ok()?` to short-circuit on errors.